policykit/integrations/discourse/tasks.py

```diff
@@ -15,21 +15,25 @@
 
 logger = logging.getLogger(__name__)
 
-def is_policykit_action(community, call_type, topic, username):
-    if username == 'PolicyKit': ## TODO: Compare IDs in future, not usernames
-        return True
-
-    current_time_minus = datetime.datetime.now() - datetime.timedelta(minutes=2)
-    logs = LogAPICall.objects.filter(
-        proposal_time__gte=current_time_minus,
-        call_type=call_type
-    )
-    if logs.exists():
-        for log in logs:
-            j_info = json.loads(log.extra_info)
-            if topic['id'] == j_info['id']:
-                return True
-    return False
+def should_create_action(community, call_type, topic, username):
+    # If topic already has an object, don't create a new object for it.
+    if DiscourseCreateTopic.objects.filter(id=topic['id']).exists():
+        return False
```

Comment:
this will have a merge conflict with https://github.com/amyxzhang/policykit/pull/300 since it creates the field `topic_id` (because https://github.com/amyxzhang/policykit/issues/332). just a heads up whoever merges in second will need to change it